pkg/resolver/search.go

```diff
@@ -410,12 +528,22 @@ func WhereClauseFilter(input *model.SearchInput) []exp.Expression {
 		for _, filter := range input.Filters {
 			if len(filter.Values) > 0 {
 				values := pointerToStringArray(filter.Values)
+				// Check if value is a number or date and get the cleaned up value
+				opDateValueMap := getOperatorAndNumDateFilter(values)
 
-				if filter.Property == "cluster" {
-					whereDs = append(whereDs, goqu.C(filter.Property).In(values).Expression())
-				} else {
-					whereDs = append(whereDs, goqu.L(`"data"->>?`, filter.Property).In(values).Expression())
+				//Sort map according to keys - This is for the ease/stability of tests when there are multiple operators
+				keys := make([]string, 0, len(opDateValueMap))
+				for k := range opDateValueMap {
+					keys = append(keys, k)
+				}
+				sort.Strings(keys)
```

Comment:
Would it be easier to store in a `[][]string` instead of  `map[string][]string`?
The `[][]string` will preserve the original order.

Reply:
Map will be easier when multiple operators are involved, to look up and check. But, we can address it later if the use case is different.